pkg/config/p2p.go

```diff
@@ -0,0 +1,20 @@
+package config
+
+// P2P holds P2P node settings.
+type P2P struct {
+	// Addresses stores the node address list in the form of "[host]:[port][:announcedPort]".
+	Addresses        []string `yaml:"Addresses"`
+	AttemptConnPeers int      `yaml:"AttemptConnPeers"`
+	// BroadcastFactor is the factor (0-100) controlling gossip fan-out number optimization.
+	BroadcastFactor int   `yaml:"BroadcastFactor"`
+	DialTimeout     int64 `yaml:"DialTimeout"`
+	MaxPeers        int   `yaml:"MaxPeers"`
+	MinPeers        int   `yaml:"MinPeers"`
+
+	PingInterval int64 `yaml:"PingInterval"`
+	PingTimeout  int64 `yaml:"PingTimeout"`
+
+	ProtoTickInterval  int64 `yaml:"ProtoTickInterval"`
+	Relay              bool  `yaml:"Relay"`
```

Comment:
This is not exactly a P2P setting, it also affects RPC.

Reply:
Fixed.

Reply:
You add it first and then remove.

Reply:
I've added it to the wrong commit, fixed.